src/main/java/com/seailz/discordjar/model/guild/Guild.java

```diff
@@ -1152,6 +1156,503 @@ public StringFormatter formatter() {
+        }
+        return GuildBan.decompile(response.body(), discordJar);
+    }
+
+    /**
+     * Bans a user from the guild. Requires the {@code BAN_MEMBERS} permission.
+     * @param userId The id of the user to ban
+     */
+    public void banUser(String userId) {
+        try {
+            new DiscordRequest(
+                    new JSONObject(),
+                    new HashMap<>(),
+                    URLS.PUT.GUILD.BAN_USER.replace("{guild.id}", id).replace("{user.id}", userId),
+                    discordJar,
+                    URLS.PUT.GUILD.BAN_USER,
+                    RequestMethod.PUT
+            ).invoke();
+        } catch (DiscordRequest.UnhandledDiscordAPIErrorException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Bans a user from the guild. Requires the {@code BAN_MEMBERS} permission.
+     * @param userId The id of the user to ban
+     */
+    public void banUser(long userId) {
+        try {
+            new DiscordRequest(
+                    new JSONObject(),
+                    new HashMap<>(),
+                    URLS.PUT.GUILD.BAN_USER.replace("{guild.id}", id).replace("{user.id}", String.valueOf(userId)),
+                    discordJar,
+                    URLS.PUT.GUILD.BAN_USER,
+                    RequestMethod.PUT
+            ).invoke();
+        } catch (DiscordRequest.UnhandledDiscordAPIErrorException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Unbans a user from the guild. Requires the {@code BAN_MEMBERS} permission.
+     * @param userId The id of the user to unban
+     */
+    public void unbanUser(String userId) {
+        try {
+            new DiscordRequest(
+                    new JSONObject(),
+                    new HashMap<>(),
+                    URLS.PUT.GUILD.BAN_USER.replace("{guild.id}", id).replace("{user.id}", userId),
+                    discordJar,
+                    URLS.PUT.GUILD.BAN_USER,
+                    RequestMethod.PUT
+            ).invoke();
+        } catch (DiscordRequest.UnhandledDiscordAPIErrorException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Unban a user from the guild. Requires the {@code BAN_MEMBERS} permission.
+     * @param userId The id of the user to unban
+     */
+    public void unbanUser(long userId) {
+        try {
+            new DiscordRequest(
+                    new JSONObject(),
+                    new HashMap<>(),
+                    URLS.PUT.GUILD.BAN_USER.replace("{guild.id}", id).replace("{user.id}", String.valueOf(userId)),
+                    discordJar,
+                    URLS.PUT.GUILD.BAN_USER,
+                    RequestMethod.DELETE
+            ).invoke();
+        } catch (DiscordRequest.UnhandledDiscordAPIErrorException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private GuildChannel getChannelById(String channelId) {
```

Comment:
get rid of this method and the related ones below

Reply:
resolved